lib/jit_preloader/active_record/associations/singular_association.rb

```diff
@@ -18,13 +18,17 @@ def load_target
           # always an N+1 query.
           record.jit_n_plus_one_tracking ||= owner.jit_n_plus_one_tracking if record
 
-          if !jit_loaded && owner.jit_n_plus_one_tracking
+          if !jit_loaded && owner.jit_n_plus_one_tracking && not_a_polymorphic_association_with_missing_type
             ActiveSupport::Notifications.publish("n_plus_one_query",
                                                  source: owner, association: reflection.name)
           end
         end
       end
     end
+    
+    private def not_a_polymorphic_association_with_missing_type
```

Comment:
This method name is quite long, what if we remove the negation from inside of the method and call it something like `is_polymorphic_without_type`

Reply:
ya... I go back and forth about this. But indeed, I dislike the name. let me think about it.